lib/wallet/local-cluster/Cardano/Wallet/Faucet.hs

```diff
@@ -2565,101 +2592,22 @@ byronIntegrationTestFunds = mconcat
             <> replicate 100 (Coin 10_000_000_000)
             <> replicate 100 (Coin 1))
 
+hwLedgerTestFunds :: [(Address, Coin)]
+hwLedgerTestFunds = do
+    mnemonic <- hwLedgerMnemonics
+    address <- take 10 $ deriveLedgerAddresses mnemonic
+    pure (address, Coin 100_000_000_000)
+  where
+    deriveLedgerAddresses :: SomeMnemonic -> [Address]
+    deriveLedgerAddresses mnemonic =
+        paymentAddress SMainnet . over icarusKey toXPub . addrXPrv
+            <$> [minBound..maxBound]
+      where
+        passphrase = mempty
+        rootXPrv = Icarus.generateKeyFromHardwareLedger mnemonic passphrase
```

Comment:
Are you aware of the difference between the icarus, ledger and trezor styles and have you verified that the generated addresses are exactly the same as the previous hard-coded ones?

It seems this line may differ:

https://github.com/cardano-foundation/cardano-wallet/blob/ba21ca0dfe8cfb47f9e9ae81f8cf37b4539b293b/lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs#L1499
https://github.com/cardano-foundation/cardano-wallet/blob/ba21ca0dfe8cfb47f9e9ae81f8cf37b4539b293b/lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs#L1521

and I'm not sure if there are more differences.

Reply:
> Are you aware of the difference between the icarus, ledger and trezor styles

As far as I have experimented with, icarus and trezor styles give rise to the same sets of addresses, the derivation isn't different;

The "ledger" style differs in 1 step only: Mnemonic -> Master private key; This is why there are 2 different mnemonic-to-master functions in the code:
`Icarus.generateKeyFromSeed` (for icarus and tresor styles) and `Icarus.generateKeyFromHardwareLedger`

>  have you verified that the generated addresses are exactly the same as the previous hard-coded ones?

There is this test `BYRON_RESTORE_09` that restores a wallet from mnemonic using the "ledger" style and it used to fail without the `hwWalletsFunds` added to the cluster; With this new derivation mechanism that I've implemented this test passes, it means that wallet recognizes 1 million ada assigned to an address derived by the faucet using the function from this PR.

Reply:
I have also verified in REPL that all the hardcoded ledger addresses are the same as derived ones.